library/bignum_mod.c

```diff
@@ -141,8 +141,15 @@ int mbedtls_mpi_mod_modulus_setup( mbedtls_mpi_mod_modulus *m,
                                    mbedtls_mpi_mod_ext_rep ext_rep,
                                    mbedtls_mpi_mod_rep_selector int_rep )
 {
-    int ret = 0;
+    int ret = MBEDTLS_ERR_MPI_BAD_INPUT_DATA;
+    size_t zero_c;
 
+    for( zero_c = 0; zero_c < p_limbs; zero_c++ )
```

Comment:
The equivalent could be achieved by

~~~~~
    /*mbedtls_mpi_uint *zp = mbedtls_calloc( p_limbs, sizeof(mbedtls_mpi_uint) );
    if( memcmp( p,zp, sizeof( mbedtls_mpi_uint ) * p_limbs ) == 0 )
    {
        mbedtls_printf("\n\t[ Minos Debug Zero Check ]\n");
        ret = MBEDTLS_ERR_MPI_BAD_INPUT_DATA;
        goto exit;
    }
    mbedtls_free(zp);*/
~~~~~

But the implemented approach is more effient and quicker since it will break the search for zero character's on the first non zero character